src/activity-logger/background/log-page-visit.test.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-env jest */
+/* eslint import/namespace: "off" */
+
+import maybeLogPageVisit from 'src/activity-logger/background/log-page-visit'
+import * as storePage from 'src/page-storage/store-page'
+import db from 'src/pouchdb'
+
+afterAll(async () => {
+    await db.destroy()
+})
+
+describe('maybeLogPageVisit', () => {
```

Comment:
As for the refactoring of this particular function, I will probably push it tomorrow or so, as I intend to use this in the release, so it might be easier for you to just rebase on that then.